### ISSUE TITLE
Include aria-label for landmarks in FSE

### ISSUE THREAD

**Issue by NicktheGeek** [WordPress/gutenberg#41412]
## What problem does this address?

WCAG requires that landmarks include aria-labels, especially if there are more than one on a page. For example, if there is a `<nav>` element in the `<header>` of the page and another `<nav>` element in the footer of the page each of those should include an aria-label like:

```
<nav aria-label="Main"></nav>
<nav aria-label="Footer"></nav>
```

Currently it is possible in FSE to update the markup of containers using landmarks, which is good, but it is not possible to easily add aria-labels to the landmarks. This is introducing an accessibility issue for all FSE sites and most users will not have any idea how to do this correctly.

## What is your proposed solution?

A new field should be added after the selector for the markup wrap. This field could conditionally show when landmarks are used or always be available. This field should include a description and/or link to some documentation explaining what the label does and how to make a good label. Ideally it should be required for any landmark, though in some cases that may be excessive like when `<main>` is used, so it is acceptable to make it optional.

**Comment by NicktheGeek** [comment]
Just a note that this does not need the block navigation label. I used the `<nav>` landmark incorrectly in my example but the Nav block does include an aria label feature. That just needs to be extended to other blocks that can have the markup switched to include landmarks.

Also, one option in the drop down is `<article>` which doesn't strictly need an aria label, but it does improve the UX in at least VO (I've yet to test NVDA or JAWS but I suspect there are modes of navigating articles which could benefit from an aria-label). In this case it should either automatically use the post title when in the loop or allow something like `%post_title%` for the label which would automatically use the post title.

**Comment by LukaszJaro** [comment]
This is creating issues in non-fse themes as well, no option to add accessible text.
